[BUG]: CUDAX cmakelists.txt should not require `CUDA` language in `project` call

### Is this a duplicate?

- [x] I confirmed there appear to be no [duplicate issues](https://github.com/NVIDIA/cccl/issues) for this bug and that I agree to the [Code of Conduct](CODE_OF_CONDUCT.md)

### Type of Bug

Compile-time Error

### Component

libcu++

### Describe the bug

I'm attempting to include `cudax` files into RMM. To do so, I've used the latest (2.7.0-rc1) tag of CCCL from github, and specified `set(CCCL_ENABLE_UNSTABLE ON)`.

This results in the `cudax` cmakelists.txt from being configured. However, that file has the line: 

```
project(cudax LANGUAGES CUDA CXX)
```

This causes a CMake configuration error because RMM does not set CUDA architectures (see below for error). This is because RMM is a header-only library that may or may not be used from `.cu` files. It's perfectly fine to allocate memory using RMM in a program or library that has no CUDA device code. 

This is the error: 

```
-- Found libcudacxx: /home/coder/rmm/build/conda/cuda-12.2/release/_deps/cccl-src/libcudacxx/lib/cmake/libcudacxx/libcudacxx-config.cmake (found suitable version "2.7.0.0", minimum required is "2.7.0.0")
-- Found Thrust: /home/coder/rmm/build/conda/cuda-12.2/release/_deps/cccl-src/thrust/thrust/cmake/thrust-config.cmake (found suitable exact version "2.7.0.0")
-- Found CUB: /home/coder/rmm/build/conda/cuda-12.2/release/_deps/cccl-src/cub/cub/cmake/cub-config.cmake (found suitable version "2.7.0.0", minimum required is "2.7.0.0")
-- Found cudax: /home/coder/rmm/build/conda/cuda-12.2/release/_deps/cccl-src/cudax/lib/cmake/cudax/cudax-config.cmake (found suitable exact version "2.7.0.0")
-- Found CCCL: /home/coder/rmm/build/conda/cuda-12.2/release/_deps/cccl-src/lib/cmake/cccl/cccl-config.cmake (found version "2.7.0.0")
-- Found Thrust: /home/coder/rmm/build/conda/cuda-12.2/release/_deps/cccl-src/thrust/thrust/cmake/thrust-config.cmake (found version "2.7.0.0")
CMake Error at /home/coder/.conda/envs/rapids/share/cmake-3.30/Modules/Internal/CMakeCUDAArchitecturesValidate.cmake:9 (message):
  CMAKE_CUDA_ARCHITECTURES:

    RAPIDS

  is not one of the following:

    * a semicolon-separated list of integers, each optionally
      followed by '-real' or '-virtual'
    * a special value: all, all-major, native

Call Stack (most recent call first):
  /home/coder/.conda/envs/rapids/share/cmake-3.30/Modules/CMakeDetermineCUDACompiler.cmake:112 (cmake_cuda_architectures_validate)
  build/conda/cuda-12.2/release/_deps/cccl-src/cudax/CMakeLists.txt:14 (project)
```

### How to Reproduce

General: Depend on `cudax` from a CMake project that does not configure CUDA architectures.



### Expected behavior

As with RMM, it is perfectly fine to use `cudax::memory_resource` and related concepts in a program or library that has no CUDA device code.

Therefore, I think the `project` line for `cudax` should be the same as the `project` line for `libcudacxx` (and RMM):

```
project(cudax LANGUAGES CXX)
```

The only binary CUDA files in `cudax` are tests, so the CUDA device / architecture configuration can be put in the cmake for the tests.

### Reproduction link

_No response_

### Operating System

_No response_

### nvidia-smi output

N/A

### NVCC version

N/A

--- a/cudax/CMakeLists.txt
+++ b/cudax/CMakeLists.txt
@@ -7,12 +7,15 @@ if (CCCL_ENABLE_CUDAX)
   cmake_minimum_required(VERSION 3.21)
 endif()
 
-project(cudax LANGUAGES CUDA CXX)
+project(cudax LANGUAGES CXX)
 
 if (NOT CCCL_ENABLE_CUDAX)
   include(cmake/cudaxAddSubdir.cmake)
   return()
 endif()
+
+# Enable CUDA only when required for the developer build, see #2609
+enable_language(CUDA)
 
 option(cudax_ENABLE_HEADER_TESTING "Test that CUDA Experimental's public headers compile." ON)
 option(cudax_ENABLE_TESTING "Build CUDA Experimental's tests." ON)

